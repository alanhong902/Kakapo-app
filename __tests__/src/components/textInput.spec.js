import React from 'react';
import TextInput from 'components/ui/textInput/textInput';
import { createComponentWithIntl } from '../helper';

function setup(props = {}) {
  const propData = {
    name: 'test',
    placeholder: 'without.translation',
    ...props,
  };
  return {
    props,
    tree: createComponentWithIntl(<TextInput {...propData} />).toJSON(),
  };
}

test('<TextInput/> render', () => {
  const { tree } = setup();
  expect(tree).toMatchSnapshot();
});

test('<TextInput/> correct value', () => {
  const { tree } = setup({ value: 42 });
  expect(tree).toMatchSnapshot();
});

test('<TextInput/> correct translation', () => {
  const { tree } = setup({ placeholder: 'nav.settings' });
  expect(tree).toMatchSnapshot();
});

test('<TextInput/> loading spinner', () => {
  const { tree } = setup({ spinner: true });
  expect(tree).toMatchSnapshot();
});
