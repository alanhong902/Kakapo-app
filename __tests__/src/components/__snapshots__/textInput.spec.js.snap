// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> correct translation 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <label
    htmlFor="test"
  >
    <span>
      Settings
    </span>
  </label>
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
</div>
`;

exports[`<TextInput/> correct value 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    defaultValue={42}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <label
    htmlFor="test"
  >
    <span>
      without.translation
    </span>
  </label>
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
</div>
`;

exports[`<TextInput/> loading spinner 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <label
    htmlFor="test"
  >
    <span>
      without.translation
    </span>
  </label>
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
  <div
    className="input-add-on-item spinner active"
  >
    <div
      className="double-bounce1"
    />
    <div
      className="double-bounce2"
    />
  </div>
</div>
`;

exports[`<TextInput/> render 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <label
    htmlFor="test"
  >
    <span>
      without.translation
    </span>
  </label>
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
</div>
`;
